engine/gamesys/src/gamesys/components/comp_model.cpp

```diff
@@ -768,9 +769,14 @@ namespace dmGameSystem
         const ModelComponent* component = render_item->m_Component;
         uint32_t material_index = render_item->m_MaterialIndex;
 
+        if (dmRender::GetBufferIndex(render_context, gfx_vertex_buffer) < world->m_DispatchCount)
+        {
+            dmRender::AddRenderBuffer(render_context, gfx_vertex_buffer);
+        }
+
         ro.Init();
         ro.m_VertexDeclaration = world->m_VertexDeclaration;
-        ro.m_VertexBuffer = (dmGraphics::HVertexBuffer) dmRender::AddRenderBuffer(render_context, gfx_vertex_buffer);
+        ro.m_VertexBuffer = (dmGraphics::HVertexBuffer) dmRender::GetBuffer(render_context, gfx_vertex_buffer);
```

Comment:
We can't advance the buffer for _every_ render object, we have to keep track on when we need a new buffer manually since multiple batch calls can be made per draw list..